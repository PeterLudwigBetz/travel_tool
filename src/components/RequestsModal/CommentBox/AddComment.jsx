
import React, { Component } from 'react';
import ImageLink from '../../image-link/ImageLink';
import Oval2 from '../../../images/Oval2.png';

class AddComment extends Component {
  render() {
    return (
      <div className="modal__modal1">
        <span className="modal__oval-copy">
          <ImageLink
            imageSrc={Oval2}
            altText="avatar"
            imageClass="modal__oval-copy" />
        </span>
        <span className="modal__add-comment">
            Add a comment
        </span>
      </div>
    );
  }
}

export default AddComment;
