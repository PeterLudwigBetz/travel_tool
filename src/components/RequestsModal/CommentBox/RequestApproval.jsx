
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ImageLink from '../../image-link/ImageLink';
import Oval2 from '../../../images/Oval2.png';

class RequestApproval extends Component {
  render() {
    const { requestData } = this.props;
    const { status } = requestData;
    return (
      <div className="modal__modal1">
        <span className="modal__mdl-icons">
          <ImageLink
            imageSrc={Oval2}
            altText="avatar"
            imageClass="modal__oval-copy" />
          <span className="modal__user-name">
            {requestData && requestData.manager}
          </span>
          <span className="modal__approval-status">
            {`${status && status.toLowerCase()} your travel request.`}
          </span>
          <span className="modal__hours-status">
           5 hours ago
          </span>
        </span>
      </div>
    );
  }
}

RequestApproval.propTypes = {
  requestData: PropTypes.object
};

RequestApproval.defaultProps = {
  requestData: {}
};

export default RequestApproval;
