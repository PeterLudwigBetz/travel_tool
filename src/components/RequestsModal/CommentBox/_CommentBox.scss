
.ql-snow.ql-toolbar button.ql-active, 
.ql-snow .ql-toolbar button.ql-active{
  color: red !important;
  background-color: #d1d0d0;
}
//works
.ql-snow.ql-toolbar button:hover, 
.ql-snow .ql-toolbar button:hover {
  background-color: rgb(158, 158, 158) !important;
}
.quill:hover {
  border: 2px solid rgba(12, 66, 245, 0.7) !important;
}

.ql-snow.ql-toolbar button svg, .ql-snow .ql-toolbar button svg{
  color: yellow !important;
}
//works
.ql-stroke {
  stroke: rgb(129, 128, 128) !important;
  stroke-width: 1;
  
}
//works
.ql-toolbar {
  border-bottom: 1px solid rgb(235, 232, 232)!important;
}
#focus-class {
  border: 2px solid rgba(21, 73, 241, 0.7);
}
// works
.quill{
  height: 142px;
  width: 80%;
  border: 0 solid rgb(235, 232, 232);
  border-radius: 2px;
  margin: 0 0 2px 0;

}
.ql-container{
  height: 100px;
  margin: 0 0 40px 0;

}
.editor {
  &__editor-form {
    height: 200px;
    width: 100%;
    margin: 0 0 40px 0;
  }
  &__btn-wrap {
    margin: 5px 0 10px 0;
  }
  &__btn-size {
    margin: 0 10px 20px 0px;
  }
  &__post-btn {
    box-sizing: border-box;
    height: 29px;
    width: 70px;
    border: 1px solid #d4d4d4;
    border-radius: 1px;
    background-color: #ffffff;
    color: #999999;
    font-family: 'DIN Pro';
    font-size: 14px;
    font-weight: 500;
    line-height: 18px;
    text-align: center;
    margin: 20px 0 20px 0;
    cursor: pointer;

    &--active {
      border: none;
      background-color: #3359db;
      color: white;
    }
  }
}
