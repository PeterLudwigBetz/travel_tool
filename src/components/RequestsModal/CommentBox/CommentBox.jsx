import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Editor } from '@tinymce/tinymce-react';
import './_CommentBox.scss';

class CommentBox extends Component {
  state = {
    dataInput: '',
    submitReady: false
  };

  handleEditorChange = event => {
    event.preventDefault();
    this.setState({
      dataInput: event.target.getContent({ format: 'text' }).trim(),
      submitReady: true
    });
  };

  handleSubmit = event => {
    event.preventDefault();
    const { handleCreateComment } = this.props;
    const { dataInput } = this.state;
    handleCreateComment(dataInput);
  };

  handleFocus = event => {
    event.target.editorContainer.style.border = '1px solid blue';
  };

  handleBlur = event => {
    event.target.editorContainer.style.border = '1px solid #E4E4E4';
  };
 

  render() {
    const { dataInput, submitReady } = this.state;
    let status = submitReady && dataInput ? '--active' : '';
    return (
      <form onSubmit={this.handleSubmit} className="editor__editor-form" id="form-id">
        <Editor
          init={{
            statusbar: false,
            plugins: 'lists',
            skin: 'lightgray',
            menubar: false, branding: false,
            toolbar: 'bold italic underline   numlist bullist   outdent indent' }}
          onChange={this.handleEditorChange}
          value={this.dataInput}
          onFocus={this.handleFocus}
          onBlur={this.handleBlur} />
        <div className="editor__btn-size">
          <span className="editor__btn-wrap">
            <button className={`editor__post-btn editor__post-btn${status} post-btn-text`} type="submit">
              Post
            </button>
          </span>
        </div>
      </form>
    );
  }
}

CommentBox.propTypes = {
  handleCreateComment: PropTypes.func.isRequired
};

export default CommentBox;
