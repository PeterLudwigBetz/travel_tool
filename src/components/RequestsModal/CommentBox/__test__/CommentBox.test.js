import React from 'react';
import { shallow } from 'enzyme';
import { CommentBox } from '../CommentBox';

// describe what we are testing
describe('Render RequestsModal component', () => {
  

  const props = {
    createComment: jest.fn(),
    handleSubmit:()=>{},
  };
  const e = {
    target: {
      editorContainer: {
        style: {
          border: ''
        }
      }
    }
  };
  const wrapper = mount(<CommentBox {...props} />);
  const wrapperInstance = wrapper.instance();

  const handles=(handle)=>{
    const spy = jest.spyOn(wrapper.instance(), handle);
    expect(spy).toHaveBeenCalledTimes(0);
    return spy;
  };
  
  const exepectCall =(spy, border) =>{
    expect(e.target.editorContainer.style.border).toBe(border);
    expect(spy).toHaveBeenCalledTimes(1);
  };
  
  // make our assertions and what we expect to happen
  
  it('should match snapshot', () => {
    const wrapper = shallow(<CommentBox {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the RequestsModal as expected', () => {
    const wrapper = shallow(<CommentBox {...props} />);
    expect(wrapper.length).toBe(1);
  });

  it('it should return comment box content as expected', () => {
    const event = {
      target: {
        innerText: 'We are travelling',
        innerHtml: 'We are travelling'
      }
    };
    const wrapper = mount(<CommentBox {...props} />);
    let { dataInput } = wrapper.instance().state;
    expect(dataInput).toEqual('<p style="color:#999999; font-size: 16px;	font-family: DIN Pro;	line-height: 20px; text-align: left; margin: 20px;">Write a comment</p>');
    wrapper.instance().handleKeyUp(event);

    dataInput = wrapper.instance().state.dataInput;
    expect(dataInput).toEqual('We are travelling');

    event.target.innerText = '';
    wrapper.instance().handleKeyUp(event);
    dataInput = wrapper.instance().state.dataInput;
    expect(dataInput).toEqual('');

  });
  
  it('it should handle onfocus as expected', () => {
    const spy = handles('handleFocus');
    wrapperInstance.handleFocus(e);
    exepectCall(spy, '1px solid blue');
  });

  it('it should handle onblur as expected', () => {
    const spy = handles('handleBlur');
    wrapperInstance.handleBlur(e);
    exepectCall(spy, '1px solid #E4E4E4');
  });

  it('should not submit if state is empty', () => {
    const event = {
      preventDefault: () => jest.fn()
    };
    const wrapper = mount(<CommentBox {...props} />);
    let form = wrapper.find('#form-id');
    form.simulate('submit');
    expect(wrapper.props().createComment).toHaveBeenCalledTimes(0);
  });

  it('should submit comment', () => {
    const event = {
      preventDefault: () => jest.fn()
    };
    const wrapper = mount(<CommentBox {...props} />);
    wrapper.state().dataInput = 'comment';
    let form = wrapper.find('#form-id');
    form.simulate('submit');
    expect(wrapper.props().createComment).toHaveBeenCalledTimes(1);
  });

  it('should handle handleEditorChange', () => {
    const spy = handles('handleEditorChange');
    wrapperInstance.handleEditorChange('this is a test comment');
    expect(spy).toHaveBeenCalledTimes(1);
  });
});
