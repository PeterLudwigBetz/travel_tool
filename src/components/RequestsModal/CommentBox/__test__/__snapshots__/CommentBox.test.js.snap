// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render RequestsModal component should match snapshot 1`] = `
<form
  className="editor__editor-form"
  id="form-id"
  onSubmit={[Function]}
>
  <Editor
    init={
      Object {
        "branding": false,
        "menubar": false,
        "plugins": "lists",
        "skin": "lightgray",
        "statusbar": false,
        "toolbar": "bold italic underline   numlist bullist   outdent indent",
      }
    }
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
  />
  <div
    className="editor__btn-size"
  >
    <span
      className="editor__btn-wrap"
    >
      <button
        className="editor__post-btn editor__post-btn post-btn-text"
        type="submit"
      >
        Post
      </button>
    </span>
  </div>
</form>
`;

exports[`Render RequestsModal component should render the onSubmit button works as exepected 1`] = `
<form
  className="editor__editor-form"
  id="form-id"
  onSubmit={[Function]}
>
  <textarea
    id="tiny-react_70794309841535684825641"
    style={
      Object {
        "visibility": "hidden",
      }
    }
  />
  <div
    className="editor__btn-size"
  >
    <span
      className="editor__btn-wrap"
    >
      <button
        className="editor__post-btn editor__post-btn post-btn-text"
        type="submit"
      >
        Post
      </button>
    </span>
  </div>
</form>
`;
